### ISSUE TITLE
B006 not raised for list comprehension

### ISSUE THREAD

**Issue by JohanSchott** [PyCQA/flake8-bugbear#184]
The code-snippet below does not raise a B006 warning (using flake8-bugbear==21.4.3).
Shouldn't B006 be raised since `a`'s default value is mutable?
```python
def f(a=[i**2 for i in range(3)]):
    a[0] += 1
    return a


print(f())
print(f())
```
